pkg/provider/azure_storageaccount.go

```diff
@@ -112,11 +112,11 @@ func (az *Cloud) getStorageAccounts(accountOptions *AccountOptions) ([]accountWi
 				}
 			}
 
-			if to.Bool(acct.IsHnsEnabled) != to.Bool(accountOptions.IsHnsEnabled) {
+			if accountOptions.IsHnsEnabled != nil && to.Bool(acct.IsHnsEnabled) != to.Bool(accountOptions.IsHnsEnabled) {
```

Comment:
`to.Bool` func would return false if `accountOptions.IsHnsEnabled` is nil, so that's not a bug.

Reply:
agreed with above. have any panic happened?

Reply:
Yes the CreateVolume unit test in the azure files csi driver panicked here.

Reply:
is that because `accountOptions` or `acct` is nil?

`to.Bool` could handle nil pointer well:
```
func Bool(b *bool) bool {
	if b != nil {
		return *b
	}
	return false
}
```

Reply:
Will check and get back.

Reply:
Turns out that in that unit test, acct.AccountProperties was nil. Have added a check for that.